apiserver/facades/client/controller/controller.go

```diff
@@ -795,20 +795,16 @@ func (c *ControllerAPI) ModifyControllerAccess(ctx context.Context, args params.
 			continue
 		}
 
-		external := !targetUserTag.IsLocal()
 		updateArgs := access.UpdatePermissionArgs{
+			Change:  permission.AccessChange(string(arg.Action)),
+			Subject: user.NameFromTag(targetUserTag),
 			AccessSpec: permission.AccessSpec{
 				Access: permission.Access(arg.Access),
 				Target: permission.ID{
 					ObjectType: permission.Controller,
 					Key:        c.controllerTag.Id(),
 				},
 			},
-			AddUser:  true,
-			External: &external,
-			ApiUser:  user.NameFromTag(c.apiUser),
```

Comment:
issue: The facade now needs to check that the apiUserName has permission sufficient to change the permissions. They need SuperUser across the controller or Admin for the entity.